src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState(Status.ALL);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+      })
+      .catch((error) => {
+        throw new Error(error.message);
```

Comment:
It's a catch block for handling errors from the server request.
You can create the state for the error message in the component and set the human-readable message via setState + render it somewhere in the document
You can set an empty value (null or an empty array) for your todos in this block
Now your app doesn't handle this error, it would fail in the case of an error

Reply:
https://github.com/mate-academy/react_dynamic-list-of-todos/pull/1710/files#r1268146192
not fixed